### DISCUSSION TITLE
Sunbird video player in 'Portrait' mode

### DISCUSSION THREAD

**Discussion by satishbagali** [Sunbird-Ed/Community#569] [Enhancement request]
Hi Team,
this is Satish from Digital Green. We are trying the 'Sunbird Video Player' on mobile app(built using Android/Kotlin). We saw few issues. They are as below. 

1. Video plays ONLY in 'Landscape' mode.  Our requirement is to play video in 'Portrait' mode as well. 
2. The video player controls are NOT functioning. For example, dragging the video tracker forward will make the video play from the beginning instead of from its new position. 
3. The '[  ]' icon at the bottom-right does NOT work. 

We would like to know if these issues are already fixed(maybe in a different version of the player)? If not, any idea when this may happen? 

Thanks,
- satish

**Comment by swayangjit** [comment]
@Satischandra Please let us know how you are using it in Android/Kotlin and which version you are using.

**Comment by Krishnaj20** [comment]
@Satischandra - Pls share additional details as requested by @swayangjit for the community to provide clarifications.

**Comment by satishgithub369** [comment]
Hi Swayangjit,
Sorry for the late response. Please find the details.

Details:
1. We are loading player on web-view in android ,by using Html , CSS and Java Script files as given in GitHub repo.
2. We are facing the same issues in web browsers like chrome. **It does not depend on the android version.** 
3. The app min version is android -8  and max is 13 

**Adding to the issues which I mentioned in my first post...**
1. In portrait mode there is no duration of the video. 
2. Side menu icon not required.
3. It has space in width. 
4. Not able to make it full screen by clicking on full screen button. 
7. Controller are not working smoothly (like seek bar)

<img width="337" alt="AndroidScnShot" src="https://github.com/Sunbird-Ed/Community/assets/138844200/c86fe90d-b4da-4941-b4c4-0457b1789543">

**Reply by swayangjit** [reply]
@satishgithub369  In the browser if you are facing this issue can you please share some logs? Also if possible share a sample app repo with the integration

**Comment by satishbagali** [comment]
Hi Sunil,
is it possible to send the logs? The question from them is pasted below.
"...In the browser if you are facing this issue can you please share some
logs? Also if possible share a sample app repo with the integration..."

The link to the SB community where I raised the question.
https://github.com/orgs/Sunbird-Ed/discussions/569

Thanks,
- satish

**Comment by sukushw** [comment]
Hi Satish,

Good Morning  😊 

Please find below zip file with sunbird player code. unzip and run html file in browser with do id. we are using same code in our android project. 
 Example : file:///Users/sunilkushwaha/Desktop/sunbird_player/sunbird.html?id=do_1138114080530104321816


[sunbird_player.zip](https://github.com/Sunbird-Ed/Community/files/12023716/sunbird_player.zip)

**Reply by swayangjit** [reply]
@sukushw  from which release you have picked this code ?

**Reply by sukushw** [reply]
@swayangjit sorry , i dont remember the release version.
  Please provide us latest release version , then we can implement and check whole the functionality of player.

**Reply by swayangjit** [reply]
@sukushw Please use this release https://github.com/Sunbird-Knowlg/sunbird-video-player/tree/release-5.5.0

**Reply by sukushw** [reply]
@swayangjit sure thank you

**Reply by sukushw** [reply]
Hi @swayangjit 
Good Morning ,

I have taken assets folder, styles.css and sunbird-video-player.js from version release-5.5.0. i found no difference on video player.
may i was using same version.
<img width="819" alt="Screenshot 2023-07-13 at 10 54 53 AM" src="https://github.com/Sunbird-Ed/Community/assets/118423804/9a95c0f3-ac1e-428e-b453-488c5c22e4a2">

**Reply by swayangjit** [reply]
@sukushw I have created a [similar ticket](https://github.com/orgs/Sunbird-Knowlg/discussions/144) in Knowledge discussion thread.

**Reply by swayangjit** [reply]
@sukushw 
@HarishGangula has commented in the thread please follow that thread.

**Reply by swayangjit** [reply]
@Satischandra @sukushw  Did you try it out using a different content. If you were not able to resolve it we can schedule a call to see what is the issue

**Reply by satishgithub369** [reply]
Hi Swayangjit, presently our Dev servers are down(for more than a week) for some reasons. So, we could NOT try. Once we get them back, we will try.

**Reply by swayangjit** [reply]
@satishgithub369  You can mock the content details response and inject the config into the player. In this case you dont need any server for the implementation. Let me know if you need any support.

**Comment by satishbagali** [comment]
Sure... I think we can have a call on this. I will check with our Android
development team about their availability and get back to you.

Regards,
- satish
